### PR TITLE
Added authentication and authorization

### PR DESCRIPTION
**Pull request by danderson00**
@mamaso When you're done with the previous PR and have time. Adds auth. 

Also fixes subtle stupidity of previous PR, i.e. mount bodyParser.json() for every request, not just the routes this module is exposing.

Sorry - just to confuse things I added copyright headers in this PR too.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "test": "npm run lint & mocha"
   },
   "dependencies": {
+    "azure-mobile-apps": "^2.1.4",
```
Comment (danderson00) [inline package.json]: This is a fairly big change. Used to consume the authenticate middleware.

File: CopyrightCheck.ps1 [hunk 2]
```diff
@@ -0,0 +1,57 @@
+param([string]$SrcRoot,
```
Comment (mamaso) [inline CopyrightCheck.ps1]: Nice, I always forget the headers

File: package.json (hunk 1 above)
Comment (mamaso) [inline package.json]: Could use this to consume the promises as well.  Unfortunately this kind of dependency ref will be ineffective when we move to 3.whatever

File: src/authorize.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            res.status(401).send('You must be logged in to use this application');
+        else
+            next();
+
+        function authenticated() {
+            return req.azureMobile && req.azureMobile.user;
+        }
+
+        function specifiesAuthorization(target) {
+            return target && (target.access === 'authenticated' || target.authorize);
```
Comment (mamaso) [inline src/authorize.js]: There isn't an easy, natural mapping for operation level auth without getting into scoping, etc, so this seems to make sense.

File: package.json (hunk 1 above)
Comment (mamaso) [inline package.json]: Does this suit our needs here better? https://docs.npmjs.com/files/package.json#peerdependencies

**Pull request merged**